sdk/core/azure-core/azure/core/pipeline/policies/_universal.py

```diff
@@ -538,7 +549,15 @@ def deserialize_from_http_generics(
         else:
             mime_type = "application/json"
 
-        return cls.deserialize_from_text(response.text(), mime_type, response=response)
+        # Rely on transport implementation to give me "text()" decoded correctly
+        return cls.deserialize_from_text(response.text(encoding), mime_type, response=response)
+
+    def on_request(self, request):
+        # type: (PipelineRequest) -> None
+        options = request.context.options
+        response_encoding = options.pop("response_encoding", self._response_encoding)
+        if response_encoding:
+            request.context.setdefault("response_encoding", response_encoding)
```

Comment:
Under what circumstance would there already be a `response_encoding` provided in the `request.context`? And wouldn't we want to replace it if the user explicitly passed on in?

Reply:
I don't see any scenario where we pop something AND it's also in the context at the same time. Context will be filled with previous pop if we pass twice in the on_request (e.g. this policy deeper than retry).
I'm fine to force it if we pop something, sounds fair behavior to me.